packages/vm/src/environment.rs

```diff
@@ -272,6 +272,10 @@ impl<A: BackendApi, S: Storage, Q: Querier> Environment<A, S, Q> {
             let func = instance.exports.get_function(name)?;
             Ok(func.clone())
         })?;
+        let function_arity = func.param_arity(store);
+        if args.len() != function_arity {
+            return Err(VmError::function_arity_mismatch());
```

Comment:
I would add function arity and number of args as additional info fields to the error for the sake of debugability when something unexpected happens. But this is nice to have and not needed for the feature.

Reply:
Done